### PR TITLE
Add tag check to release draft creation

### LINKED ISSUE #3725: Creating a Release allowed to be created with a invalid tag name with spaces
- Gitea version (or commit ref): 1.4
- Git version: 2.15.1
- Operating system: Centos7 amd64
- Database (use `[x]`):
  - [x ] PostgreSQL 10
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x ] No, URL returns error
  - [ ] Not relevant
- Log gist:
DeleteReleaseByID: git tag -d: exec(18:DeleteReleaseByID (git tag -d): 3) failed: exit status 1() stdout: stderr: error: Tag '0.3 a' nicht gefunden. - error: Tag '0.3 a' nicht gefunden.

## Description
- Create a new release with a new tag and use a tag name with a space "0.3 a"
- Now we have a release draft, based on a non existing tag, that can't be deleted.

## Screenshots
tagwithspace

tagwithspace-editrelease

tagwithspace-deleterelease

### PR DESCRIPTION
**Pull request by Bwko**
Fixes #3725

### CODE REVIEW

File: models/release.go [hunk 1]
```diff
@@ -140,11 +140,35 @@ func createTag(gitRepo *git.Repository, rel *Release) error {
...
+		if !isValidTag(rel.TagName) {
+			return ErrInvalidTagName{rel.TagName}
+		}
 		rel.CreatedUnix = util.TimeStampNow()
 	}
 	return nil
 }
 
+//  See https://git-scm.com/docs/git-check-ref-format for more details
+func isValidTag(tag string) bool {
```
Comment (lafriks) [inline models/release.go]: We already have ref-name validation: https://github.com/go-gitea/gitea/blob/f3833b7ce4dc78095194808c6e07d8ae133e7ab5/modules/validation/binding.go#L38

I think that `rel.TagName` just needs to be added form validation binding

**Review by lafriks — CHANGES_REQUESTED** [review]

File: models/release.go (hunk 1 above)
Comment (Bwko) [reply models/release.go]: Ugh, I've completely missed that binding :sweat:

File: modules/auth/repo_form.go [hunk 2]
```diff
@@ -365,7 +365,7 @@ func (f *MergePullRequestForm) Validate(ctx *macaron.Context, errs binding.Error
 
 // NewReleaseForm form for creating release
 type NewReleaseForm struct {
-	TagName    string `binding:"Required"`
+	TagName    string `binding:"Required;GitRefName"`
```
Comment (lunny) [inline modules/auth/repo_form.go:374]: but https://github.com/go-gitea/gitea/blob/master/templates/repo/release/new.tmpl#L20 the form name is `tag_name`.

**Review by lunny — CHANGES_REQUESTED** [review]

**Comment by jonasfranz** [comment]
LGTM

**Review by daviian — APPROVED** [review]

**Comment by jonasfranz** [comment]
Needs @lafriks and @lunny review

**Review by lunny — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**